framework/test/unit/modules/token/cc_method.spec.ts

```diff
@@ -69,104 +80,69 @@ describe('CrossChain Forward command', () => {
-		tokenInteropMethod = new TokenInteroperableMethod(
-			tokenModule.stores,
-			tokenModule.events,
-			tokenMethod,
-		);
-		interopMethod = {
-			getOwnChainAccount: jest.fn().mockResolvedValue({ id: Buffer.from([0, 0, 0, 1]) }),
-			send: jest.fn(),
-			error: jest.fn(),
-			terminateChain: jest.fn(),
-			getChannel: jest.fn().mockResolvedValue({ messageFeeTokenID: defaultTokenID }),
-		};
-		// const minBalances = [
-		// 	{ tokenID: defaultTokenIDAlias, amount: BigInt(MIN_BALANCE) },
-		// 	{ tokenID: defaultForeignTokenID, amount: BigInt(MIN_BALANCE) },
-		// ];
-		tokenMethod.addDependencies(interopMethod as never);
-		// tokenInteropMethod.addDependencies(interopMethod);
-		// tokenMethod.init({
-		// 	ownchainID: Buffer.from([0, 0, 0, 1]),
-		// 	minBalances,
-		// });
+		tokenInteropMethod = new TokenInteroperableMethod(tokenModule.stores, tokenModule.events);
+		tokenInteropMethod.init(ownChainID);
+		tokenInteropMethod.addDependencies({
+			send: jest.fn().mockResolvedValue(true),
+			getMessageFeeTokenID: jest.fn().mockResolvedValue(defaultTokenID),
+		} as never);
 
-		stateStore = new PrefixedStateReadWriter(new InMemoryPrefixedStateDB());
 		methodContext = createMethodContext({
-			stateStore,
-			eventQueue: new EventQueue(0),
+			stateStore: new PrefixedStateReadWriter(new InMemoryPrefixedStateDB()),
+			eventQueue: new EventQueue(0).getChildQueue(Buffer.from([0])),
 		});
 		userStore = tokenModule.stores.get(UserStore);
-		await userStore.save(methodContext, defaultAddress, defaultTokenIDAlias, defaultAccount);
+		await userStore.save(methodContext, defaultAddress, defaultTokenID, defaultAccount);
 		await userStore.save(methodContext, defaultAddress, defaultForeignTokenID, defaultAccount);
 
 		const supplyStore = tokenModule.stores.get(SupplyStore);
-		await supplyStore.set(methodContext, defaultTokenIDAlias.slice(CHAIN_ID_LENGTH), {
+		await supplyStore.set(methodContext, defaultTokenID, {
 			totalSupply: defaultTotalSupply,
 		});
 
-		const escrowStore = tokenModule.stores.get(EscrowStore);
+		escrowStore = tokenModule.stores.get(EscrowStore);
 		await escrowStore.set(
 			methodContext,
-			Buffer.concat([
-				defaultForeignTokenID.slice(0, CHAIN_ID_LENGTH),
-				defaultTokenIDAlias.slice(CHAIN_ID_LENGTH),
-			]),
+			Buffer.concat([defaultForeignTokenID.slice(0, CHAIN_ID_LENGTH), defaultTokenID]),
 			{ amount: defaultEscrowAmount },
 		);
 		await escrowStore.set(
 			methodContext,
-			Buffer.concat([Buffer.from([3, 0, 0, 0]), defaultTokenIDAlias.slice(CHAIN_ID_LENGTH)]),
+			Buffer.concat([Buffer.from([3, 0, 0, 0]), defaultTokenID]),
 			{ amount: defaultEscrowAmount },
 		);
 	});
 
-	// TODO: Update with https://github.com/LiskHQ/lisk-sdk/issues/7577
-	describe.skip('beforeApplyCCM', () => {
-		it('should reject if fee is negative', async () => {
+	describe('beforeCrossChainCommandExecution', () => {
+		it('should credit fee to transaction sender if token id is not native', async () => {
```

Comment:
even if token ID is native, it is crediting to transaction sender / relayer?

Reply:
yes, I added this condition as a seperate test (number 3 in this scope) `should deduct escrow account for fee and credit to sender if token id is native` but I can also combine both if you prefer.